src/AzureStore.cs

```diff
@@ -71,30 +74,27 @@ public sealed class AzureStore : AzureReadOnlyStore, IAzureStore
         /// </remarks>
         public Uri GetSignedUploadUrl(string name, TimeSpan life)
         {
-            var blob = Staging.GetBlockBlobReference(name);
-            var token = blob.GetSharedAccessSignature(new SharedAccessBlobPolicy
-            {
-                Permissions = SharedAccessBlobPermissions.Write | SharedAccessBlobPermissions.Delete,
-                SharedAccessExpiryTime = new DateTimeOffset(DateTime.UtcNow + life)
-            },
-            new SharedAccessBlobHeaders
-            {
-                CacheControl = "private"
-            });
+            var blob = Staging.GetBlobClient(name);
+            var url = blob.GenerateSasUri(BlobSasPermissions.Write | BlobSasPermissions.Delete,
+                expiresOn: new DateTimeOffset(DateTime.UtcNow + life));
+
 
-            return new Uri(blob.Uri.AbsoluteUri + token);
+            return url;
         }
 
         /// <summary> Compress a blob into the archive container and set its tier to "archive" in Azure. </summary>
         /// <param name="blob"> The blob to be archived. </param>
         public async Task ArchiveBlobAsync(IAzureReadBlobRef blob, CancellationToken cancel = default)
         {
             var aBlob = await blob.GetBlob();
-            var destinationCloudBlockBlob = Archive.GetBlockBlobReference(aBlob.Name);
+            var destinationBlobClient = Archive.GetBlobClient(aBlob.Name);
 
-            if (await destinationCloudBlockBlob.ExistsAsync(cancel)) return;
+            if (await destinationBlobClient.ExistsAsync(cancel)) return;
 
-            using (var azureWriteStream = await destinationCloudBlockBlob.OpenWriteAsync(cancel))
+            using (var azureWriteStream = await destinationBlobClient.OpenWriteAsync(true, new BlobOpenWriteOptions
+            {
+                BufferSize = 1024
```

Comment:
@VictorNicollet Does it seem like a good default to you?

Reply:
I think 4MB (`4 * 1024 * 1024`) is a better default given the typical data volume.